Applications/spire/include/spire/ui/window.hpp

```diff
@@ -47,6 +47,10 @@ namespace Spire {
       void set_svg_icon(const QString& icon_path,
         const QString& unfocused_icon_path);
 
+      void resize(int width, int height);
```

Comment:
Why do you need these methods instead of handing events? This is unfortunately not a robust way of dealing with resizes, similar thing for for ```setFixedSize```.

Reply:
Do you have an idea for how this would work? I can't use the resize event because it results in infinite recursion.

Reply:
Think about potential ways of avoiding infinite recursion. Ask yourself why does infinite recursion happen, and what bookkeeping you can use to recognize that situation and immediately converge to a fixed point.

Reply:
I can keep a bool as a flag but it feels kind of dirty. There's nothing inherent about a resize event I can rely on to determine this.

Reply:
What happens if you remove those methods entirely?

Reply:
I was trying to make it so each window implementation (Book View, T&S, e.g.,) wouldn't have to know the size of the title bar, which is why I re-implemented the resize() function. This would work without it but if the title bar height changed every window implementation would have to be updated.

I can probably remove the setFixedSize() implementation if I set the window's resizable status in the constructor.

For the original question, if I simply removed them then the window implementations wouldn't have the correct initial size, and fixed size windows would have a maximize button.

Reply:
What about making the title bar's height a public static member? Then the window implementation's can use that for calculating their initial height.

Reply:
Can you provide a minimal example where this resize method is needed? Just some pseudocode that shows the issue?

Reply:
Somewhere like the BookViewWindow's constructor:

resize(scale(220, 410));

Without the reimplemented resize I would have to manually account for the height of the title bar.